states/florida/main.py

```diff
@@ -1,24 +1,35 @@
-import json
-import glob
-import os
-
-from common import dir_path
-
-
-files = glob.glob(dir_path(__file__) + '/cache/*.json')
-data = []
-print(f'Found {len(files)} files')
-for file in files:
-  with open(file) as fh:
-    datum = json.load(fh)
-    county = datum['title'].split('Supervisor')[0].strip()
-    data += [{
-      'locale': county,
-      'official': datum['name'].replace(u'\xa0', ' ').split(',')[0].strip(),
-      'emails': [datum['email'].split(':')[1].strip()],  # ignore leading 'mailto:'
-      'url': datum['url'],
-      'county': county,
-    }]
-
-with open('public/florida.json', 'w') as fh:
-  json.dump(data, fh)
+from bs4 import BeautifulSoup
+
+from common import dir_path, cache_request, decode_email, diff_and_save
+
+BASE_URL = 'https://dos.elections.myflorida.com/supervisors/'
+
+
+def parse_county(soup, county_link):
+  datum = {}
+  datum['locale'] = soup.find('p', class_='title').text.split('Supervisor')[0].strip()
+  datum['official'] = soup.find('span', class_='bigRed').text.replace(u'\xa0', ' ').split(',')[0].strip()
+  links = soup.find(id='rightContent')('a')
+  datum['emails'] = [decode_email(links[0].find('span', class_='__cf_email__').get('data-cfemail')).strip()]
+  datum['url'] = links[1]['href'].strip()
+  datum['county'] = datum['locale']
+  return datum
+
+if __name__ == '__main__':
+  data = []
+
+  text = cache_request(BASE_URL)
+  soup = BeautifulSoup(text, 'html.parser')
+
+  # links to county pages
+  county_links = map(lambda x: x['href'], soup.select('a[href^=countyInfo]'))
+  for county_link in county_links:
+    text = cache_request(BASE_URL+county_link)
+    data.append(parse_county(BeautifulSoup(text, 'html.parser'), county_link))
+  print("Florida # of counties: {}".format(len(data)))
+
+  # sort by locale for consistent ordering
+  data.sort(key=lambda x: x['locale'])
+
+  diff = diff_and_save(data, 'public/florida.json')
+  print("Florida diff: {}".format(diff))
```

Comment:
I would just add the print to the `diff_and_save` with an optional flag to the function to print the diff.